lib/session-picker.coffee

```diff
@@ -0,0 +1,112 @@
+{SelectListView} = require 'atom-space-pen-views'
+_ = require 'lodash'
+
+Config = require './config'
+services = require('./jupyter-js-services-shim')
+uuid = require 'uuid'
+
+class CustomListView extends SelectListView
+    initialize: (@emptyMessage, @onConfirmed) ->
+        super
+        @addClass('watch-language-picker')
+        @storeFocusedElement()
+        @panel ?= atom.workspace.addModalPanel(item: this)
+        @panel.show()
+        @focusFilterEditor()
+
+    getFilterKey: ->
+        'name'
+
+    destroy: ->
+        @cancel()
+
+    viewForItem: (item) ->
+        element = document.createElement('li')
+        element.textContent = item.name
+        element
+
+    cancelled: ->
+        @panel?.destroy()
+        @panel = null
+        @editor = null
+
+    confirmed: (item) ->
+        @onConfirmed?(item)
+        @cancel()
+
+    getEmptyMessage: ->
+        @emptyMessage
+
+module.exports =
+class SessionPicker
+    constructor: (onChosen) ->
+        @_onChosen = onChosen
+
+    toggle: ->
+        @_path = atom.workspace.getActiveTextEditor().getPath() + '-' + uuid.v4()
+        gatewayListing = new CustomListView('No gateways available', @onGateway.bind this)
+        gatewayListing.setLoading 'Loading gateways...'
+        items = Config.getJson('gateways')
+        if items?
```

Comment:
`if not _.isEmpty items` would work here since `Config.getJson` returns an empty array on default.